### PR TITLE
Added support for auth using WorkloadIdentityCredential

### PR DESCRIPTION
**Pull request by ociappara**

### CODE REVIEW

**Comment by ociappara** [comment]
Still working on testing end to end code changes

File: assemblyline/odm/models/config.py [hunk 1]
```diff
@@ -189,25 +188,30 @@ class OAuthProvider(odm.Model):
...
     client_id: str = odm.Optional(odm.Keyword(),
-                                  description="ID of your application to authenticate to the OAuth provider")
+                                description="ID of your application to authenticate to the OAuth provider")
     client_secret: str = odm.Optional(odm.Keyword(),
-                                      description="Password to your application to authenticate to the OAuth provider")
+                                    description="Password to your application to authenticate to the OAuth provider")
+    auto_no_secret: bool = odm.Boolean(default=False, description="Should we use the client secret for the OAuth?")
+    client_scope: str = odm.Optional(odm.Keyword(),
+                                    description="Managed Identity scope to authenticate to the OAuth provider")
+    client_tenent: List[str] = odm.List(odm.Keyword(), default=[],
```
Comment (cccs-rs) [inline assemblyline/odm/models/config.py:211]: Typo: client_**tenants**

Comment (ociappara) [reply assemblyline/odm/models/config.py]: Fixed.

**Comment by ociappara** [comment]
Are we okay to merge this to Dev ?

**Comment by cccs-sgaron** [comment]
Merging into dev as : https://github.com/CybercentreCanada/assemblyline-base/pull/1722

**Review by cccs-sgaron — APPROVED** [review]

**Pull request merged**